### ISSUE TITLE
HTTP 422 Unprocessable Entity

### ISSUE THREAD

**Issue by MeikelLP** [mirego/accent#113]
I'm currently trying to translate Unity3D into my native language. Unity uses the gettext format. Translation files are ~1800KB and ~40,000 lines.

When I try to upload those files I get a `HTTP 422 Unprocessable Entity` with the response `file cannot be parsed`.

The file is called `de.po`.

Any workaround for this? It's not because of the size.

Here's the [code line](https://github.com/mirego/accent/blob/51ceec7cafb4c7a59055e28e8e9843dbf4a93a9b/lib/web/plugs/movement_context_parser.ex#L105)

It's not even working with the example on the website
```
## `msgid`s in this file come from POT (.pot) files.
msgid ""
msgstr ""
"Language: en;"
"Plural-Forms: nplurals=2; plural=n == 1 ? 0 : 1;"

msgid "can't be blank"
msgstr "can’t be blank"
```

**Comment by olafura** [comment]
I also had that error, in my case it was because of duplications of msgid.

This might help you:
```
diff --git a/lib/langue/formatter/gettext/parser.ex b/lib/langue/formatter/gettext/parser.ex
index 51974ab..534ed1d 100644
--- a/lib/langue/formatter/gettext/parser.ex
+++ b/lib/langue/formatter/gettext/parser.ex
@@ -6,6 +6,7 @@ defmodule Langue.Formatter.Gettext.Parser do
 
   def parse(%{render: render, document: document}) do
     {:ok, po} = Gettext.PO.parse_string(render)
+    |> IO.inspect()
     entries = parse_translations(po)
     top_of_the_file_comment = join_string(po.top_of_the_file_comments)
     header = join_string(po.headers)
```

**Comment by MeikelLP** [comment]
I actually tried to do it with a single translation today and that didn't work as well.

**Comment by olafura** [comment]
I tried it with your example and it worked, at least uploading the file. But then again, I just started evaluation this project ;)

**Comment by MeikelLP** [comment]
Hm, that's strange. I can't read/ understand Erlang/elexir. 
Can you shortly explain what's happening in that code block and what could cause this issue?

**Comment by olafura** [comment]
So the parser is pretty strict, which is somewhat understandable. So if it hits any error you out of luck. Ideally you would communicate the problems to the user since they have the line number that are problematic.

If you have access to Elixir try doing this in the the cloned directory:
```
iex -S mix
iex> Gettext.PO.parse_file("file.po")
```
You should get the problematic areas and you only have to get it right one time unless you are using something to generate it.

**Comment by MeikelLP** [comment]
Can you try it? It's an edited Unity3D translation file. I edited it with Crowdin.io and worked perfectly. 

https://send.firefox.com/download/f81794395990b6f9/#wfzYdZ5_e3PE-w1SzOXZyg

This link expires after 100 downloads or 7 days. No secrets included.

**Comment by olafura** [comment]
{:error, 29038, "found duplicate on line 23 for msgid: 'Edit'"}

**Comment by MeikelLP** [comment]
Okay, that's strange - because it didn't even work with one string ... I'll try again.

**Comment by olafura** [comment]
There are more issues

**Comment by MeikelLP** [comment]
Same kind?

**Comment by olafura** [comment]
Yeah, I'm wondering if I should patch Gettext so it can ignore duplicates.

**Comment by MeikelLP** [comment]
Crowdin and Unity itself can handle it. I mean it might be "useful" if you respect the reference. But still a bad user behaviour if you use a key twice for different things.

**Comment by olafura** [comment]
I'll look into it from the Gettext side because this seems to be common enough for people and they are valid files.

**Comment by MeikelLP** [comment]
So what's going to happen now?

**Comment by olafura** [comment]
Found the issue:
https://github.com/elixir-lang/gettext/issues/199

**Comment by MeikelLP** [comment]
I'd love to see a fix for that^^

The only workaround for now is removing duplicates am I right?

**Comment by olafura** [comment]
I've already started trying that out here:
https://send.firefox.com/download/6d9441be67ffe863/#AS-lpihVE-lfXaYQSx-z2Q

**Comment by MeikelLP** [comment]
I have to re-setup accent and will try it later. But I downloaded it for now.

Sadly the workaround is not a full fix for me because as I already said the context does matter :/

**Comment by olafura** [comment]
Of course

**Comment by MeikelLP** [comment]
So https://github.com/elixir-lang/gettext/issues/199 got fixed. When will this fix be available in Accent?

CC @simonprev

**Comment by simonprev** [comment]
Fixed by #118
